src/ui/public/kbn_top_nav/kbn_top_nav_controller.js

```diff
@@ -65,8 +65,8 @@ export default function ($compile) {
         run: (item) => this.toggle(item.key)
       }, opt);
 
-      defaultedOpt.hideButton = isFunction(opt.hideButton) ? opt.hideButton : () => opt.hideButton;
-      defaultedOpt.disableButton = isFunction(opt.disableButton) ? opt.disableButton : () => opt.disableButton;
+      defaultedOpt.hideButton = isFunction(opt.hideButton) ? opt.hideButton : () => !!opt.hideButton || false;
```

Comment:
These defaults are not necessary because the `!!` forces the value to be either `true` or `false`.